Feat/update lcobucci jwt

marko-ilic opened the pull request.

healerz commented on tests/unit/Helpers/Tokens/IdTokenVerifierTest.php:
```diff
@@ -120,26 +131,7 @@ public function testThatTokenWithMissingAudFails()
-        );
-    }
-
-    public function testThatTokenWithNonStringOrArrayAudFails()
```
I think this is a valid test case. Or do I miss something?

marko-ilic replied:
Now aud is always array

healerz replied:
yes this test case doesn't make sense if it gets converted in the parser 👍

healerz commented:
can you do me the favour and already write comments in this PR, that we would want to add to the "official" pr?

marko-ilic commented on src/Helpers/Tokens/TokenVerifier.php:
```diff
@@ -110,23 +110,19 @@ public function verify(string $token, array $options = []) : array
         }
 
-        if (is_array($tokenAud) && ! in_array($this->audience, $tokenAud)) {
+        if (! in_array($this->audience, $tokenAud)) {
```
new parser will convert it to an array
https://github.com/lcobucci/jwt/blob/058b61b376339208622b6dc9d38b70a122338c20/src/Token/Parser.php#L103

healerz replied:
ah I see 👍

marko-ilic commented on src/Helpers/Tokens/TokenVerifier.php:
```diff
@@ -110,23 +110,19 @@ public function verify(string $token, array $options = []) : array
             );
         }
 
-        if (is_array($tokenAud) && ! in_array($this->audience, $tokenAud)) {
```
The new parser will convert 'aud' claim to an array so is_array check is not necessary anymore and elseif part too because $tokenAud can not be string
https://github.com/lcobucci/jwt/blob/058b61b376339208622b6dc9d38b70a122338c20/src/Token/Parser.php#L103

marko-ilic commented on src/Helpers/Tokens/TokenVerifier.php:
```diff
@@ -136,12 +132,13 @@ public function verify(string $token, array $options = []) : array
-        if (! $tokenExp || ! is_int($tokenExp)) {
+        $tokenExp = $verifiedToken->claims()->get('exp', false);
+
+        if (! $tokenExp instanceof \DateTimeImmutable) {
```
After parsing the token 'exp' and 'iat' will be the instance of DateTimeImmutable

healerz commented:
@marko-ilic maybe we can also add a comment explaining why we added the firebase/jwt library as a dev dependency. Including our suggestion to not validate the token in this library at all

marko-ilic commented:
> @marko-ilic maybe we can also add a comment explaining why we added the firebase/jwt library as a dev dependency. Including our suggestion to not validate the token in this library at all

I added a comment for firebase/jwt 
`The package firebase/php-jwt is added as dev dependency, because with the new lcobucci/jwt it is not possible to create all necessary non-valid tokens in functional tests.`

I'll add a suggestion for not validate token 👍
